src/main/java/org/folio/helper/PurchaseOrderLineHelper.java

```diff
@@ -303,34 +289,22 @@ private Future<Void> updateEncumbranceStatus(CompositePoLine compOrderLine, PoLi
-    return StringUtils.equals(poLine.getPaymentStatus().value(), CompositePoLine.PaymentStatus.CANCELLED.value())
-      && (StringUtils.equals(compOrderLine.getPaymentStatus().value(), CompositePoLine.PaymentStatus.AWAITING_PAYMENT.value())
-        || StringUtils.equals(compOrderLine.getPaymentStatus().value(), CompositePoLine.PaymentStatus.PARTIALLY_PAID.value())
-        || StringUtils.equals(compOrderLine.getPaymentStatus().value(), CompositePoLine.PaymentStatus.ONGOING.value()));
+  private boolean isUnreleasedEncumbrances(PoLine compOrderLine, PoLine poLine) {
+    return StringUtils.equals(poLine.getPaymentStatus().value(), PoLine.PaymentStatus.CANCELLED.value())
+      && (StringUtils.equals(compOrderLine.getPaymentStatus().value(), PoLine.PaymentStatus.AWAITING_PAYMENT.value())
+        || StringUtils.equals(compOrderLine.getPaymentStatus().value(), PoLine.PaymentStatus.PARTIALLY_PAID.value())
+        || StringUtils.equals(compOrderLine.getPaymentStatus().value(), PoLine.PaymentStatus.ONGOING.value()));
   }
 
-  /**
-   * Handles update of the order line depending on the content in the storage. Returns {@link Future} as a result.
-   * In case the exception happened in future lifecycle, the caller should handle it. The logic is like following:<br/>
-   * 1. Handle sub-objects operations's. All the exception happened for any sub-object are handled generating an error.
-   * 2. Store PO line summary. On success, the logic checks if there are no errors happened on sub-objects operations and
-   * returns succeeded future. Otherwise {@link HttpException} will be returned as result of the future.
-   *
-   * @param compOrderLine The composite {@link CompositePoLine} to use for storage data update
-   * @param lineFromStorage {@link JsonObject} representing PO line from storage (/acq-models/mod-orders-storage/schemas/po_line.json)
-   */
-  public Future<Void> updateOrderLine(CompositePoLine compOrderLine, JsonObject lineFromStorage, RequestContext requestContext) {
-    return purchaseOrderLineService.updateSearchLocations(compOrderLine, requestContext)
-      .compose(v -> purchaseOrderLineService.updatePoLineSubObjects(compOrderLine, lineFromStorage, requestContext))
-      .compose(poLine -> purchaseOrderLineService.updateOrderLineSummary(compOrderLine.getId(), poLine, requestContext))
-      .onFailure(throwable -> logger.error("PoLine with id - '{}' partially updated but there are issues processing some PoLine sub-objects", compOrderLine.getId()))
-      .mapEmpty();
+  public Future<Void> updateOrderLineInStorage(PoLine poLine, RequestContext requestContext) {
+    return purchaseOrderLineService.updateSearchLocations(poLine, requestContext)
+      .compose(v -> restClient.put(resourceByIdPath(PO_LINES_STORAGE, poLine.getId()), poLine, requestContext))
+      .onFailure(throwable -> logger.error("Error saving poLine with id - '{}'", poLine.getId()));
```

Comment:
better to add throwable to log, without it we can not know the reason of failure

Reply:
OK